### ISSUE TITLE
Contours shifted from grayscale image

### ISSUE THREAD

**Issue by paramoreta** [aplpy/aplpy#311]
Hi:

So, I am trying to plot this image (galaxy.fits) and overlay its **own** contours.

The header keywords A_0_0 and alike existed in the original fits but, when trying to load the file, aplpy gave me a KeyError saying they could not be found. Weird, because they were in the header and had actual numbers on them.
I tried removing them and re-running astrometry. Then aplpy complained no more. However, when trying to plot the image (greyscale) and its contours (in colour, same image) , I get a funny shift that I do not know where it comes from or why.

I simply do this:

```
image = "galaxy.fits"

im = aplpy.FITSFigure(image, north=True, figsize=(16,16), auto_refresh=True)

im.show_contour(image, levels=[50.6, 50.7, 50.8, 50.9, 51.0, 51.1 ,52])
im.show_grayscale(vmin=49.5, vmax=52, stretch='log')
im.add_colorbar()
```

and I get an orientation-ok image but it the contours are **shifted** from the grayscale image, as you can see in `north.png` image. 

If I try removing the parameter `north=True`, I get no shift, but the orientation is messed up, as you can see in `no-north.png` image.

Any light on why am I getting the shift with the `north=True` version?

Thanks in advance!

Unexplainable **shift** between contours and image (north=True); **north.png** image:
![north](https://cloud.githubusercontent.com/assets/5229049/18581868/b9d66be8-7c02-11e6-8e2a-82e7dd0648bf.png)

Messed up orientation but **no shift** between contours and grayscale (without north=True parameter) **no-north.png** image (no shift!):
![no-north](https://cloud.githubusercontent.com/assets/5229049/18581869/b9d7dc76-7c02-11e6-93c9-459659dc1bdb.png)

**Comment by paramoreta** [comment]
Any help or input in this contours-image mismatch, please?

**Comment by astrofrog** [comment]
@paramoreta - sorry for the delay - could you send me the FITS file for this, either by uploading it and putting a link here, or if needed, emailing it to thomas.robitaille@gmail.com?

**Comment by paramoreta** [comment]
No problem about the delay. Thanks for answering, @astrofrog.
I am afraid I cannot upload it here or anywhere, so I'll just email it.
And... sent!

**Comment by paramoreta** [comment]
Hey there, Thomas. Any news regarding this one? Cheers mate.

**Comment by astrofrog** [comment]
@paramoreta - sorry for the long delay. I am suspecting this could be related to distortions. The FITS file you sent has issues, because many keywords appear twice or more in the header. For example CTYPE1 appears twice, A_0_0 appears four times. Once this is fixed, I think the issue could be that there are instances of: ``wcs_world2pix`` and ``wcs_pix2world`` in APLpy that should be ``all_world2pix`` and ``all_pix2world`` in order to work correctly with distortions. Could you try fixing these instances to see if this helps in your case?

**Comment by paramoreta** [comment]
@astrofrog - Thanks for the feedback. Your suspicions were in the right direction.

I found out some time ago that the repetition of the polynomial coefficients A_0_0 (as well as A_0_1, A_0_2, ... , B_0_0, B_0_1, ...) was causing this and some other handling/visualization problems.

Despite RA and DEC had unique keywords, the (unexpected) extra sets of coefficients were residues from different WCS calculations that, instead of being **replaced** by the new calculations, were being **added** as new keywords in the header with the same name. Some commands were reading who-knows what set of keywords, messing up with the visualization and some operations. Thanks for the help!

TL;DR, the repetition of the polynomial coefficients A_0_0 (and family) keywords in the FITS file header cause visualization problems as well as some others. Keeping only one set of keywords should do the trick.
